### PR TITLE
Allow cache to handle 5XX errors and retry 5 times

### PR DESCRIPTION
**Pull request by foursixnine**

### CODE REVIEW

**Comment by foursixnine** [comment]
The hax0r in me of late 90's likes this PR.

File: lib/OpenQA/Worker/Cache.pm [hunk 1]
```diff
@@ -129,6 +129,16 @@ sub download_asset {
             $asset = undef;
         }
     }
+    elsif ($tx->res->code && $tx->res->code =~ /5[0-9]{2}/) {
```
Comment (kraih) [inline lib/OpenQA/Worker/Cache.pm]: You can use the `is_server_error` method for this. http://mojolicious.org/perldoc/Mojo/Message/Response#is_server_error

File: lib/OpenQA/Worker/Cache.pm [hunk 2]
```diff
@@ -176,6 +186,16 @@ sub get_asset {
             next;
         }
         $ret = download_asset($job->{id}, lc($asset_type), $asset, ($result->{etag}) ? $result->{etag} : undef);
+        if ($ret =~ /5[0-9]{2}/ && $n--) {
```
Comment (kraih) [inline lib/OpenQA/Worker/Cache.pm]: Couldn't the path returned by `download_asset` contain a substring `5xx`? If plain numbers are safe perhaps anchoring the regex (`/^5[0-9]{2}$/`) would work.

File: lib/OpenQA/Worker/Cache.pm (hunk 1 above)
Comment (foursixnine) [inline lib/OpenQA/Worker/Cache.pm]: is_server_error did the trick, however no error code is provided for situations where the the webserver is down. I used [531](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#Cloudflare) for that :)

File: lib/OpenQA/Worker/Cache.pm (hunk 2 above)
Comment (foursixnine) [inline lib/OpenQA/Worker/Cache.pm]: It does :)

**Pull request merged**